### PR TITLE
HEEDLS-273 Initial Menu page customisation back end

### PR DESCRIPTION
**Pull request by benjimarshall**
## Changes
Add CourseSettings field to course content service

## Testing
Amend tests

### CODE REVIEW

File: DigitalLearningSolutions.Data.Tests/Services/CourseContentServiceTests.cs [hunk 1]
```diff
@@ -47,7 +47,8 @@ public void Get_course_content_of_partially_complete_course_should_return_course
                 true,
                 85,
                 90,
-                0
+                0,
+                null
```
Comment (FridaTveit) [inline DigitalLearningSolutions.Data.Tests/Services/CourseContentServiceTests.cs:51]: Probably worth having at least one test with a course which has settings, either by changing which course is used for one of these tests or by adding a new test? 🙂

**Review by FridaTveit — COMMENTED** [review]
One comment and as discussed on slack it might be worth doing the parsing in the model. Otherwise looks good 🙂

File: DigitalLearningSolutions.Data.Tests/Services/CourseContentServiceTests.cs (hunk 1 above)
Comment (benjimarshall) [reply DigitalLearningSolutions.Data.Tests/Services/CourseContentServiceTests.cs:51]: Yeah I agree. I'm going to add that parsing now and put tests for that here 🙂

Comment (benjimarshall) [reply DigitalLearningSolutions.Data.Tests/Services/CourseContentServiceTests.cs:52]: There are no active courses that have a non-null course settings for the fields used by the initial menu. Instead I've added a test that modifies a course to have some interesting settings, and then test that the `GetCourseContent` method gets the correct settings back 🙂

**Comment by benjimarshall** [comment]
The course settings JSON parser has been added in 06dfe77 🙂

File: DigitalLearningSolutions.Data/Models/CourseSettings.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
+﻿namespace DigitalLearningSolutions.Data.Models
+{
+    using System.Collections.Generic;
+    using Newtonsoft.Json;
+
+    public class CourseSettings
+    {
+        public readonly bool ShowPercentage = true;
```
Comment (FridaTveit) [inline DigitalLearningSolutions.Data/Models/CourseSettings.cs:8]: Minor but using a property with getter only is more consistent with other classes I think 🙂

**Review by FridaTveit — COMMENTED** [review]
One minor comment otherwise looks good 🙂

File: DigitalLearningSolutions.Data/Models/CourseSettings.cs (hunk 2 above)
Comment (benjimarshall) [reply DigitalLearningSolutions.Data/Models/CourseSettings.cs]: Yep, sorry! Fixed in ea68e57 🙂

**Review by FridaTveit — APPROVED** [review]
Looks good 🙂

**Pull request merged**